model_table_has_many.go

```diff
@@ -172,5 +172,11 @@ func indirectAsKey(field reflect.Value) interface{} {
 	if field.IsNil() {
 		return nil
 	}
+
+	if valuer, ok := field.Interface().(driver.Valuer); ok {
+		v, _ := valuer.Value() // No need to record logs, will panic later
```

Comment:
Do we need to be compatible with cases similar to #1107? 
Suppose the user is using *ULID,  if we don’t handle this specifically, 
it will panic because [16]byte is unhashable in Go.

Reply:
We only deal with pointer-based `driver.Valuer` implementation. What do you think?

> https://go.dev/play/p/kY2YJUrEx8r

Reply:
In your test case, the struct itself is not comparable, but the result returned by Value is comparable.
So it would have panicked before, but now it won’t.

In the example I provided, the struct(array) itself is comparable, 
but the result returned(slice) by Value is not comparable.
So it wouldn’t have panicked before, but now it will.
```go
type ULID [16]byte

func (id *ULID) Scan(any) error               { return nil }
func (id *ULID) Value() (driver.Value, error) { return []byte{}, nil }

type User struct {
	ID    *ULID `bun:",pk"`
	Name  string
	Posts []*Post `bun:"rel:has-many,join:id=user_id"`
}

type Post struct {
	PostID string `bun:",pk"`
	UserID *ULID
}
```

Maybe I misunderstood, but this isn’t related to whether it’s pointer-based, is it?

Reply:
Interfaces like `driver.Valuer`, `json.Marshaler`, and most other encoding interfaces are generally implemented directly on values rather than pointers.

Protobuf generated code has another characteristic, [`DoNotCopy`](https://github.com/protocolbuffers/protobuf-go/blob/master/internal/pragma/pragma.go#L29) , in addition to [`DoNotCompare`](https://github.com/protocolbuffers/protobuf-go/blob/master/internal/pragma/pragma.go#L22) , this imposes a strict requirement that all methods must be pointer-based when extending the generated code.

Deviating from this best practice will inevitably lead to more issues. For example, just like now.

I will fix it later.